NVIDIA-Merlin/Transformers4Rec
# Fix the predictions returned by `Trainer.predict(..)`

Fixes #636

### Goals :soccer:
- Fix the predictions returned by `Trainer.predict()`.
- Ensure `predict_top_k` can only be applied to NextItemPredictionTask. 
- Fix the output shapes of the model's predictions in the sequential binary classification or the regression task.

### Implementation Details :construction:


### Testing Details :mag:
- Add `test_sequential_binary_classification_model` to test a sequential model trained with a per-item binary classification task. 
- Checks the shapes of predictions in `test_trainer_music_streaming`
- Add `test_trainer_trop_k_with_wrong_task` to ensure the BC task cannot be used with `predict_top_k > 0`
